account_vat_period_end_statement/models/account.py

```diff
@@ -735,23 +747,27 @@ def _get_credit_debit_lines(self, statement):
             ]
         )
         for tax in taxes:
-            # se ho una tassa padre con figli cee_type, condidero le figlie
-            if any(
-                tax_ch
-                for tax_ch in tax.children_tax_ids
-                if tax_ch.cee_type in ("sale", "purchase")
+            if (
+                tax.vat_statement_account_id.id in statement.account_ids.ids
+                or not statement.account_ids
```

Comment:
Invertirei l'`or`. Usi .ids sul recordset di una relazione e poi testi se è vuoto. Mi pare meglio fare il contrario.

```
if not statement.account_ids or tax.vat_statement_account_id.id in statement.account_ids.ids:
```

Poi (ma è solo questione stilistica) io preferisco evitare di aggiungere livelli di indentazione, specie con un `if` che copre l'intero blocco di una `for`, in genere preferisco esplicitare una condizione di 'skip':
```
for tax in taxes:
    if statement.account_ids and tax.vat_statement_account_id not in statement.account_ids:
        continue
```
oppure filtrare prima di entrare nel loop:
```
if statement.account_ids:
    taxes = taxes.filtered(lambda tax: tax.vat_statement_account_id in statement.account_id)

for tax in taxes:
...
```
che in questo caso tra tutte le alternative mi sembra la piu leggibile (e come side effect, riduce il diff, specie la parte di indentazione diversa che riechiede di controllare a mano se non ci sono differenze nel codice).

Infine tra recordset si possono usare gli operatori `in` e `not in` direttamente, senza passare da `x.id in y.ids`